tests/test_project.py

```diff
@@ -669,6 +682,797 @@ def helper_test_list_thermal_maps(project):
+            ),
+            coordinate_units="in",
+            image_bounds=ImageBounds(-4.2551, -4.4217, 7.0277, 9.2132),
+            legend_bounds=LegendBounds(4.1811, -3.365, 4.7947, 0.3478),
+            legend_orientation=LegendOrientation.VERTICAL,
+            max_temperature=47.0,
+            max_temperature_units="C",
+            min_temperature="30.0",
+            min_temperature_units="C",
+        )
+        thermal_map_files = [
+            {
+                "file_name": "Thermal Image.jpg",
+                "file_type": ThermalMapsFileType.IMAGE,
+                "file_comment": "Update",
+                "thermal_board_side": ThermalBoardSide.BOTTOM,
+                "file_data": file_data,
+                "thermal_profiles": ["Environmental/1 - Temp Cycle - Min"],
+                "cca_names": ["Main Board"],
+            }
+        ]
+        project.update_thermal_maps("Tutorial Project", thermal_map_files)
+        pytest.fail("No exception raised when using an invalid parameter")
+
+    except SherlockUpdateThermalMapsError as e:
+        assert str(e.str_itr()) == (
+            "['Update thermal maps error: " "Invalid minimum temperature for thermal map 0.']"
+        )
+
+    try:
+        file_data = ImageFile(
+            board_bounds=BoardBounds(
+                [(-3.7464, -2.2515), (3.7464, -2.141), (3.6159, 2.2395), (-3.5724, 2.2487)]
+            ),
+            coordinate_units="in",
+            image_bounds=ImageBounds(-4.2551, -4.4217, 7.0277, 9.2132),
+            legend_bounds=LegendBounds(4.1811, -3.365, 4.7947, 0.3478),
+            legend_orientation=LegendOrientation.VERTICAL,
+            max_temperature=47.0,
+            max_temperature_units="C",
+            min_temperature=30.0,
+            min_temperature_units=0,
+        )
+        thermal_map_files = [
+            {
+                "file_name": "Thermal Image.jpg",
+                "file_type": ThermalMapsFileType.IMAGE,
+                "file_comment": "Update",
+                "thermal_board_side": ThermalBoardSide.BOTTOM,
+                "file_data": file_data,
+                "thermal_profiles": ["Environmental/1 - Temp Cycle - Min"],
+                "cca_names": ["Main Board"],
+            }
+        ]
+        project.update_thermal_maps("Tutorial Project", thermal_map_files)
+        pytest.fail("No exception raised when using an invalid parameter")
+
+    except SherlockUpdateThermalMapsError as e:
+        assert str(e.str_itr()) == (
+            "['Update thermal maps error: " "Invalid minimum temperature units for thermal map 0.']"
+        )
+
+    if not project._is_connection_up():
+        return
+
+    try:
+        project.update_thermal_maps("Tutorial Project", [])
+        pytest.fail("No exception raised when using an invalid parameter")
+    except Exception as e:
+        assert type(e) == SherlockUpdateThermalMapsError
```

Comment:
The above test doesn't require being connected to Sherlock to fail.
The test to validate the exception type on a request failure needs to ensure that the request is sent to Sherlock and Sherlock returns a response indicating a failure.
See helper_test_delete_project() for an example. In this test, the project name is set so that it refers to a project that doesn't exist.